rule.go

```diff
@@ -51,8 +51,8 @@ func NewRule() *Rule {
 		SuppressIfgroup:   -1,
 		SuppressPrefixlen: -1,
 		Priority:          -1,
-		Mark:              -1,
-		Mask:              -1,
+		Mark:              0,
+		Mask:              0xFFFFFFFF,
```

Comment:
Thinking about this some more, it seems that one issue with this approach is that it is not possible to represent the `0x0/0xFFFFFFFF` mark/mask pair, as this is what we use for the defaults (and we use it to mean "match any mark").

I cannot thing of a solution for this that does not involve making at least one of these 2 fields a pointer (`*uint32`).
For example, with a `uint32` `Mark` and a `*uint32` `Mask`, the following logic could be used:
* if `Mark == 0` and `Mask == nil`, omit both attributes
* else if `Mark != 0` and `Mask == nil`, set the mark attribute but not the mask attribute
* else set both attributes

cc @aboch

Reply:
I fell into the trap of thinking that -1 == 0xFFFFFFFF, so I missed the 0x0/0xFFFFFFFF mark/mask pair.
I have made the changes and added some necessary tests in rule_test.go. Please review again.